ApiReview/API.Polly.Core/NoDocs/Polly.Simmy.Outcomes/OutcomeStrategyOptions.TResult.cs

```diff
@@ -0,0 +1,15 @@
+// Assembly 'Polly.Core'
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Polly.Simmy.Outcomes;
+
+public class OutcomeStrategyOptions<TResult> : MonkeyStrategyOptions
+{
+    public Func<OnOutcomeInjectedArguments<TResult>, ValueTask>? OnOutcomeInjected { get; set; }
+    public Func<OutcomeGeneratorArguments, ValueTask<Outcome<TResult>?>>? OutcomeGenerator { get; set; }
+    public Outcome<TResult>? Outcome { get; set; }
```

Comment:
Hide this property? 

This will cause problems with disposable outcomes such as `HttpResponseMessage`. Outcome strategy will return pre-defined outcome and caller will dispose it this making the `Outcome` invalid for all other calls. 

For simple scenarios people can use this extension:

``` 
public static ResiliencePipelineBuilder<TResult> AddChaosResult<TResult>(this ResiliencePipelineBuilder<TResult> builder, bool enabled, double injectionRate, TResult result)
```

And if using options:

``` csharp
new ResiliencePipelineBuilder<HttpResponseMessage>()
    .AddChaosResult(new()
    {
        OutcomeGenerator = _ =>
        {
            var outcome = Outcome.FromResult(new HttpResponseMessage(HttpStatusCode.InternalServerError));

            return new ValueTask<Outcome<HttpResponseMessage>?>(outcome);
        }
    })
    .Build();
```

@clpun, I know we had this issue in our code when using older Simmy and I am thinking if we can prevent this with new design.

Reply:
This is done, don't know why I can't mark comments as resolved 🤔